checkpoint/ddb/retryer_test.go

```diff
@@ -0,0 +1,32 @@
+package ddb
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+func TestRetyer(t *testing.T) {
+	var r DefaultRetryer
+	retryableError := awserr.New(dynamodb.ErrCodeProvisionedThroughputExceededException, "error is retryable", errors.New("don't care what is here"))
+
+	p := &r
+	p = nil
+	shouldRetry := p.ShouldRetry(retryableError)
+	if shouldRetry != false {
```

Comment:
Isn't it supposed to be true since the error should be retried?

Reply:
p is set to nil. ShouldRetry accepts a pointer receiver, so you can do whatever you want to that receiver. So instead of panic on nil pointer returns false, and can log it too.